### DISCUSSION TITLE
Questions regarding the widgets

### DISCUSSION THREAD

**Discussion by Joe23232** [elkowar/eww#310] [Q&A] [answered]
Hi, I have some questions regarding the widgets:

1. Am I able to code my own widgets using `eww`?
2. Am I able to make something like this:
![Extra blue](https://user-images.githubusercontent.com/34926497/137583184-53e206cc-ad3c-410b-8187-a9fad58ddb05.png)

**Comment by elkowar** [comment, marked as answer]
Generally yes, you can write your own widgets - although there of course are some limitations. Most of this _should_ be possible, although not all that easily.

**Reply by Joe23232** [reply]
I see thanks mate.

Would I be writing my own widgets in Rust using this framework?

**Reply by elkowar** [reply]
No, widgets are currently just implemented in terms of preexisting (GTK based) widgets. If you have any general purpose widgets that you are missing, you're more than welcome to implement them and contribute them to eww, though ^^

I am considering implementing some sort of actual plugin API that would allow for user-implemented, writen-in-rust widgets in the future, but I fear that is still quite far away from becoming a reality

**Reply by Joe23232** [reply]
@elkowar Thanks mate.
